<?xml version="1.0"?>
<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.1" />

  <node pkg="loam" type="multiScanRegistration" name="multiScanRegistration" output="screen">
    
    <!-- options: VLP-16  HDL-32  HDL-64E -->
    <!-- <param name="lidar" value="VLP-16" /> -->
    <!-- <param name="lidar" value="HDL-32" /> -->
    <param name="lidar" value="HDL-64E" />

    <param name="scanPeriod" value="$(arg scanPeriod)" />

    <remap from="/multi_scan_points" to="/velodyne_points" />
    <remap from="/imu/data" to="/raw_imu" />
    <!-- <remap from="/imu/data" to="/djiros/imu" />   -->
  </node>

  <node pkg="loam" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam" type="laserMapping" name="laserMapping" output="screen">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam" type="transformMaintenance" name="transformMaintenance" output="screen">
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam)/rviz_cfg/loam_velodyne.rviz" />
  </group>

</launch>
